modules/arm_plugin/src/arm_plugin.cpp

```diff
@@ -103,6 +128,10 @@ QueryNetworkResult Plugin::QueryNetwork(const CNNNetwork& network, const ConfigM
     if (model == nullptr) {
          IE_THROW() << "Arm Plugin supports only ngraph cnn network representation";
     }
+    auto valid = CheckStatic(model);
+    if (!std::get<0>(valid)) {
+        IE_THROW() << "Arm Plugin doesn't support inputs having dynamic shapes. Dynamic inputs are: " << std::get<1>(valid);
+    }
```

Comment:
I suppose QueryNetwork should report lack of dynamic shapes support via missing such operations in resulting QueryNetworkResult object.

Reply:
The original error `get_shape was called on a descriptor::Tensor with dynamic shape` is raised in `Plugin::Transform` method that called before operation validation. I believe, we can't address your comment.